### PR TITLE
Add some doc for export_for_training

### PR DESCRIPTION
**Pull request by tugsbayasgalan**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #135918
* #135912
* #135080

Differential Revision: [D62610491](https://our.internmc.facebook.com/intern/diff/D62610491)

### CODE REVIEW

**Comment by tugsbayasgalan** [comment]
@tugsbayasgalan has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D62610491).

**Comment by tugsbayasgalan** [comment]
@tugsbayasgalan has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D62610491).

**Comment by tugsbayasgalan** [comment]
@tugsbayasgalan has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D62610491).

**Comment by tugsbayasgalan** [comment]
@tugsbayasgalan has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D62610491).

File: docs/source/export.rst [hunk 1]
```diff
@@ -291,6 +291,176 @@ torch.export>`), but seeing as the context manager does not affect the tensor
 computations in the model, we can go with the non-strict mode's result.
 
 
+.. _Training Export:
+
+Training Export
```
Comment (avikchaudhuri) [inline docs/source/export.rst:296]: Not a good title. Maybe something like "Export for Training and Inference"?

File: docs/source/export.rst [hunk 2]
```diff
@@ -291,6 +291,176 @@ torch.export>`), but seeing as the context manager does not affect the tensor
...
+^^^^^^^^^^^^^^^^^
+
+In PyTorch 2.5, we introduced a new API called :func:`export_for_training`.
+It's still going through hardening, so if you run into any issues, please file
+them to Github with the "oncall: export" tag.
+
+In this API, we produce the most generic IR that contains all ATen operators
+(including both functional and non-functional) which can be used to train in
+eager PyTorch Autograd. This API is intended for PT2 quantization training use cases
```
Comment (avikchaudhuri) [inline docs/source/export.rst:305]: I'm not sure why you need to mention quantization training. E.g., you'd want this for normal eager training too.

File: docs/source/export.rst [hunk 3]
```diff
@@ -291,6 +291,176 @@ torch.export>`), but seeing as the context manager does not affect the tensor
...
+and will soon be the default IR of torch.export.export in the near future. To read further about
```
Comment (avikchaudhuri) [inline docs/source/export.rst:306]: soon vs. near future, pick one

File: docs/source/export.rst [hunk 4]
```diff
@@ -291,6 +291,176 @@ torch.export>`), but seeing as the context manager does not affect the tensor
...
+        output_specs=[
+            OutputSpec(kind=<OutputKind.BUFFER_MUTATION: 3>, arg=TensorArgument(name='getitem_3'), target='bn.running_mean'),
+            OutputSpec(kind=<OutputKind.BUFFER_MUTATION: 3>, arg=TensorArgument(name='getitem_4'), target='bn.running_var'),
+            OutputSpec(kind=<OutputKind.BUFFER_MUTATION: 3>, arg=TensorArgument(name='add'), target='bn.num_batches_tracked'),
+            OutputSpec(kind=<OutputKind.USER_OUTPUT: 1>, arg=TensorArgument(name='getitem'), target=None)
+        ]
+    )
+    Range constraints: {}
+
+Here you can see that, we kept `conv2d` op in the IR while decomposing the rest. Now the IR is an functional IR
```
Comment (avikchaudhuri) [inline docs/source/export.rst:411]: Edited:

Here you can see that we kept `conv2d` op in the IR while decomposing the rest. Now the IR is a functional IR
containing core aten operators except for `conv2d`.

You can do even more customization by directly registering your chosen decomposition behaviors.

File: docs/source/export.rst [hunk 5]
```diff
@@ -291,6 +291,176 @@ torch.export>`), but seeing as the context manager does not affect the tensor
...
+            ]
+        )
+    Range constraints: {}
+
+
+From the above output, you can see :func:`export_for_training` produces pretty much the same `ExportedProgram`
+as :func:`export` except for the operators in the graph. You can see that we captured `batch_norm` in the most general
+form. This op is non-functional and will be lowered to different ops when running under inference mode.
+
+You can also go from this IR to an inference IR via :func:`run_decompositions` with arbitrary customizations
```
Comment (avikchaudhuri) [inline docs/source/export.rst:367]: Edited:

From the above output, you can see that :func:`export_for_training` produces pretty much the same `ExportedProgram`
as :func:`export` except for the operators in the graph. You can see that we captured `batch_norm` in the most general
form. This op is non-functional and will be lowered to different ops when running inference.

You can also go from this IR to an inference IR via :func:`run_decompositions` with arbitrary customizations.

File: docs/source/export.rst [hunk 6]
```diff
@@ -291,6 +291,176 @@ torch.export>`), but seeing as the context manager does not affect the tensor
...
+them to Github with the "oncall: export" tag.
+
+In this API, we produce the most generic IR that contains all ATen operators
+(including both functional and non-functional) which can be used to train in
+eager PyTorch Autograd. This API is intended for PT2 quantization training use cases
+and will soon be the default IR of torch.export.export in the near future. To read further about
+the motivation behind this change, please refer to
+https://dev-discuss.pytorch.org/t/why-pytorch-does-not-need-a-new-standardized-operator-set/2206
+
+With this API, and :func:`run_decompositions()`, you should be able to get any inference IR with
```
Comment (avikchaudhuri) [inline docs/source/export.rst:310]: Edited:

When this API is combined with :func:`run_decompositions()`, you should be able to get inference IR with
any desired decomposition behavior.

**Review by avikchaudhuri — APPROVED** [review]
Looks good, suggested edits for grammar / style.

**Comment by tugsbayasgalan** [comment]
@tugsbayasgalan has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D62610491).

**Comment by tugsbayasgalan** [comment]
@tugsbayasgalan has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D62610491).

**Comment by tugsbayasgalan** [comment]
@tugsbayasgalan has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D62610491).

**Comment by tugsbayasgalan** [comment]
@tugsbayasgalan has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D62610491).

**Comment by tugsbayasgalan** [comment]
@tugsbayasgalan has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D62610491).

**Comment by tugsbayasgalan** [comment]
@tugsbayasgalan has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D62610491).

**Comment by tugsbayasgalan** [comment]
@tugsbayasgalan has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D62610491).

**Pull request closed without merging**